private/rules/pom_file.bzl

```diff
@@ -1,33 +1,41 @@
 load(":has_maven_deps.bzl", "MavenInfo", "has_maven_deps")
 
-_PLAIN_DEP = """        <dependency>
-            <groupId>{0}</groupId>
-            <artifactId>{1}</artifactId>
-            <version>{2}</version>
-        </dependency>"""
+def _format_dep(unpacked):
+    return "".join([
+ "        <dependency>\n",
+ "            <groupId>%s</groupId>\n" % unpacked.groupId,
+ "            <artifactId>%s</artifactId>\n" % unpacked.artifactId,
+ "            <version>%s</version>\n" % unpacked.version,
+("            <type>%s</type>\n" % unpacked.type) if unpacked.type and unpacked.type != "jar" else "",
+("            <scope>%s</scope>\n" % unpacked.scope) if unpacked.scope and unpacked.scope != "compile" else "",
+ "        </dependency>",
+    ])
 
-_TYPED_DEP = """        <dependency>
-            <groupId>{0}</groupId>
-            <artifactId>{1}</artifactId>
-            <version>{2}</version>
-            <type>{3}</type>
-        </dependency>"""
+def _unpack_coordinates(coords):
+    """Takes a maven coordinate and unpacks it into a struct with fields
+    groupId, artifactId, version, type_, scope
```

Comment:
`type_`? A quick exploration suggests you'll be fine with `type` on it's own:

```starlark
foo = struct(type = "bar")
print("Foo is: %s and dir is %s" % (foo, dir(foo)))
```

prints `Foo is: struct(type = "bar") and dir is ["to_json", "to_proto", "type"]`

Reply:
You may want to mark those fields as code (eg. `groupId` rather than just groupId) so it renders more legibly in the generated docs.

Reply:
used `type_` just because vim syntax highlighted it as a keyword, but maybe it works, let's try